modules/signoz/main.tf

```diff
@@ -36,6 +36,42 @@ spec:
   - repoURL: 'https://github.com/Sage-Bionetworks-Workflows/eks-stack.git'
     targetRevision: ${var.git_revision}
     ref: values
+  %{if var.enable_otel_ingress}
+  - repoURL: 'https://github.com/Sage-Bionetworks-Workflows/eks-stack.git'
+    targetRevision: ${var.git_revision}
+    path: modules/signoz/resources-otel-ingress
+    kustomize:
+      patches:
+      - target:
+          kind: ReferenceGrant
+        patch: |-
+          - op: replace
+            path: /spec/from/0/namespace
+            value: ${var.gateway_namespace}
+      - target:
+          kind: HTTPRoute
+        patch: |-
+          - op: replace
+            path: /metadata/namespace
+            value: ${var.gateway_namespace}
+          - op: replace
+            path: /spec/rules/0/backendRefs/0/namespace
+            value: ${var.namespace}
+      - target:
+          kind: SecurityPolicy
+        patch: |-
+          - op: replace
+            path: /metadata/namespace
+            value: ${var.gateway_namespace}
+          - op: replace
+            path: /spec/jwt/providers
+            value:
+              - name: auth0
+                remoteJWKS:
+                  uri: ${var.auth0_jwks_uri}
+                audiences:
+                  - ${var.cluster_name}-telemetry

```

Comment:
Adding this security policy let's us do filtering at 2 levels:

1. When a request enters the cluster the gateway is checking that a JWT is provided before passing it to any routes
2. This SecurityPolicy affects the route for the pod to verify the audience is as expected


Testing the export a data with a valid JWT that does not have the expected audience fails with an HTTP 403:
`ERROR:opentelemetry.exporter.otlp.proto.http.trace_exporter:Failed to export batch code: 403, reason: Audiences in Jwt are not allowed`

Reply:
Moved to https://github.com/Sage-Bionetworks-Workflows/eks-stack/pull/35/files#r1830011397